### ISSUE TITLE
Not work date format

### ISSUE THREAD

**Issue by olegstan** [vitalets/x-editable#746]
Hi.
When i set format it's not work, i always get date in format "Wed Apr 15 2015 04:00:00 GMT+0400 (Московское время (зима))"

$('.editable-estimated_resolution_at-js').editable({
        type: 'date',
        format: 'yyyy-mm-dd',
        viewformat: 'yyyy-mm-dd',
        emptytext: 'укажите дату',
        url: laroute.route('request_edit_in_place'),
        success: function (response, newValue) {
            $.ajax({
                url: laroute.route('request_edit_in_place'),
                method: 'POST',
                dataType: 'JSON',
                data: {
                    field: 'estimated_resolution_at',
                    request_id: this.getAttribute('data-request-id'),
                    value: newValue
                },
                success: function (response) {
                },
                error: function (response) {

                }
            });
        }
    });

**Comment by fabioperreca** [comment]
also me have the same problem

**Comment by Alex-Sokolov** [comment]
Work for me with:
```js
$('.date-editable').editable({
	type: 'date',
	validate: function (value) {
		if (!moment(value).isValid()) {
			return 'Введите корректную дату';
		}
	},
	format: 'dd.mm.yyyy',
	tpl: ' <div class="input-group date"><input class="form-control" type="text"><span class="input-group-addon"><i class="fa fa-calendar"></i></span></div>',
	datepicker: {
		weekStart: 1,
		language: 'ru'
	}
});
```

**Comment by fabioperreca** [comment]
hi,
i've tried but still not work. My code:

    $("#div_id").editable({
          type: 'date',
          validate: function (value) {
            console.log(value);
              if (!moment(value).isValid()) {
                  return 'Введите корректную дату';
              }
          },
          format: 'dd.mm.yyyy',
          tpl: ' <div class="input-group date"><input class="form-control" type="text"><span class="input-group-addon"><i class="fa fa-calendar"></i></span></div>',
          datepicker: {
              weekStart: 1,
      }});

variable 'value' in validate function return Date {Wed May 27 2015 02:00:00 GMT+0200 (CEST)} and not dd.mm.yyyy


Do you apply .editable({...})  to a <a> tag or <span> ?

**Comment by Alex-Sokolov** [comment]
That's correct. Value stored Date in long format as you wrote. In my example I use moment.js for easier validating that entered date is valid: `moment(value).isValid()`

I'm apply to dynamic amount of fields with that class btw.

**Comment by fabioperreca** [comment]
i can see you example?
thanks

**Comment by Alex-Sokolov** [comment]
All you need - convert value to format which you want.
Look:
```js
validate: function (value) {
	console.log(value.format("YYYY-MM-DD")); // 2010-05-04
},
```

**Comment by grapchomcreative** [comment]
I'm getting the same date response as Olegstan mentioned at the beginning of this post. I'm using AJAX and the date is storing correctly in the PHP, so my problem has be how we are handling the success in the Javascript. I'm trying to display the newly selected date to the user. 

I have attached a screenshot below to show you how the return is displaying. 

`$(".edit-scan-date").editable({
        url: "/api/edit_scan_date.php",
        ajaxOptions: {
            dataType: "json",
            cache: !1
        },
        type: 'date',
        viewformat: 'YYYY-MM-DD',     
        combodate: {
            minYear: 2000,
            maxYear: 2050,
            minuteStep: 1
        },
        success: function(e, t) { 
            console.log(t); 
            console.log('Date:: '+t['_d']);
            console.log('e.value:: '+e.value);
            return t['_d'];
        }`

![editable-screenshot](https://user-images.githubusercontent.com/7140141/51397408-29210500-1b0f-11e9-9e26-aaf299b294ec.jpg)

**Comment by eslym** [comment]
2021 is ending and this critical issue still exists
